src/profile/components/multiItemRow/MultiItemRow.tsx

```diff
@@ -0,0 +1,122 @@
+import { TextInput } from 'hds-react';
+import React from 'react';
+import { Field, Formik, FormikProps, Form } from 'formik';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+
+import styles from './multiItemRow.module.css';
+import { EditDataValue, EmailValue, PhoneValue } from '../../helpers/editData';
+import { createFormFieldHelpers } from '../../helpers/formik';
+import { phoneSchema, emailSchema } from '../../../common/schemas/schemas';
+import FormButtons from '../formButtons/FormButtons';
+import EditButtons from '../editButtons/EditButtons';
+import commonFormStyles from '../../../common/cssHelpers/form.module.css';
+import SaveIndicator from '../saveIndicator/SaveIndicator';
+import { useCommonEditHandling } from '../../hooks/useCommonEditHandling';
+import { RowItemProps } from '../multiItemEditor/MultiItemEditor';
+import { getFormFields } from '../../helpers/formProperties';
+
+type EmailAndPhoneFormikValue = { value: string };
+
+function MultiItemRow(props: RowItemProps): React.ReactElement {
+  const { data, testId, dataType } = props;
+  const { t } = useTranslation();
+  const {
+    isEditing,
+    currentAction,
+    actionHandler,
+    isNew,
+  } = useCommonEditHandling(props);
+  const { value, primary } = data;
+  const schema = dataType === 'phones' ? phoneSchema : emailSchema;
+  const inputValue: string =
+    (dataType === 'phones'
+      ? (value as PhoneValue).phone
```

Comment:
It's little hard to follow quickly where value comes from. Would maybe prefer applying full path like: `props.data.value` that shows it right away that it comes from props. Or then try to destructure it only once from props.

Reply:
Doesn't this apply to all destructured variables? So never use it? I a bit lost how to solve this and make it clearer:
```
const { data, testId, dataType } = props;
...
const { value, primary } = data;
```

Reply:
Yeah it probably does 😅 . I just think that if we get a variable by destructuring multiple times at different rows, it makes it little harder to read where it originally came from. Is it possible to get it by destructuring it only once? Or then just apply the props.data.value directly

Reply:
Changed to const { data:{ value , primary }, testId, dataType } = props;